src/script_var_handler.rs

```diff
@@ -145,23 +145,43 @@ impl Drop for ScriptVarHandler {
     }
 }
 
-/// Run a command in sh, returning its stdout-handle wrapped in a
-/// [`BufReader`]. If running the command fails, will print a warning
-/// and return `None`.
-fn try_run_command(command: &str) -> Option<Child> {
-    let result = std::process::Command::new("sh")
-        .arg("-c")
-        .arg(command)
-        .stdout(Stdio::piped())
-        .stderr(Stdio::inherit())
-        .stdin(Stdio::null())
-        .spawn();
-
-    match result {
-        Ok(handle) => Some(handle),
-        Err(err) => {
-            eprintln!("WARN: Error running command from script-variable: {:?}", err);
-            None
+#[derive(Debug)]
+struct TailVarProcess {
+    pid: nix::unistd::Pid,
+    out_fd: filedescriptor::FileDescriptor,
+}
+
+impl TailVarProcess {
+    pub fn run(command: &str) -> Result<Self> {
+        use nix::unistd::*;
+
+        let pipe = filedescriptor::Pipe::new()?;
+
+        match unsafe { fork()? } {
+            ForkResult::Child => {
+                std::mem::drop(pipe.read);
+                dup2(pipe.write.as_raw_fd(), std::io::stdout().as_raw_fd())?;
+                setpgid(Pid::from_raw(0), Pid::from_raw(0))?;
+                execv(
+                    CString::new("/bin/sh")?.as_ref(),
+                    &[CString::new("sh")?, CString::new("-c")?, CString::new(command)?],
+                )?;
+                unreachable!("Child fork called exec, thus the process was replaced by the command the user provided");
+            }
+            ForkResult::Parent { child, .. } => {
+                std::mem::drop(pipe.write);
+                setpgid(child, child)?;
+                Ok(TailVarProcess {
+                    pid: child,
+                    out_fd: pipe.read,
+                })
+            }
+        }
+    }
+
+    pub fn kill(self) {
+        unsafe {
+            nix::libc::kill(self.pid.as_raw(), libc::SIGTERM);
```

Comment:
nix::sys::signal::kill takes the Pid newtype and is prolly what you should use here.